controllers/apim/ratelimitpolicy_controller.go

```diff
@@ -117,480 +102,269 @@ func (r *RateLimitPolicyReconciler) Reconcile(eventCtx context.Context, req ctrl
+				Name: limitadorRatelimitsName(rlpKey, i+1),
+				// Currently, Limitador Operator (v0.2.0) will configure limitador services with
+				// RateLimit CRs created in the same namespace.
+				Namespace: common.KuadrantNamespace,
+			},
+			Conditions: rlSpec.Conditions,
+			MaxValue:   rlSpec.MaxValue,
+			Namespace:  rlSpec.Namespace,
+			Variables:  rlSpec.Variables,
+			Seconds:    rlSpec.Seconds,
 		}
 
-		// remove the parentRef entry on filters patch
-		if err := r.removeParentRefEntry(ctx, wasmEnvoyFilter, owner); err != nil {
-			logger.Error(err, "failed to remove parentRef entry on the ratelimit filters patch")
+		ratelimit := ratelimitfactory.RateLimit()
+		err := r.ReconcileResource(ctx, &v1alpha1.RateLimit{}, ratelimit, alwaysUpdateRateLimit)
+		if err != nil && !apierrors.IsAlreadyExists(err) {
+			logger.Error(err, "ReconcileResource failed to create/update RateLimit resource")
 			return err
 		}
-		logger.Info("successfully deleted/updated ratelimit filters patch")
 	}
-
-	logger.Info("successfully detached RateLimitPolicy from specified gateways and hosts")
+	logger.Info("successfully created/updated RateLimit resources")
 	return nil
 }
 
-func (r *RateLimitPolicyReconciler) attachToNetwork(ctx context.Context, gateways []string, owner string, rlp *apimv1alpha1.RateLimitPolicy) error {
+func (r *RateLimitPolicyReconciler) reconcileNetworkResourceBackReference(ctx context.Context, rlp *apimv1alpha1.RateLimitPolicy) error {
 	logger := logr.FromContext(ctx)
-	logger.Info("Attaching RateLimitPolicy to a network")
-
-	for _, gw := range gateways {
-		gwKey := common.NamespacedNameToObjectKey(gw, rlp.Namespace)
-		gwLabels := gatewayLabels(ctx, r.Client(), gwKey)
-
-		// create/update ratelimit filters patch
-		// fetch already existing filters patch or create a new one
-		wasmEnvoyFilterKey := client.ObjectKey{Namespace: gwKey.Namespace, Name: rlFiltersPatchName(gwKey.Name)}
-		wasmEnvoyFilter := &istio.EnvoyFilter{}
-		if err := r.Client().Get(ctx, wasmEnvoyFilterKey, wasmEnvoyFilter); err != nil {
-			if !apierrors.IsNotFound(err) {
-				logger.Error(err, "failed to get ratelimit filters patch")
-				return err
-			}
-		}
+	httpRoute, err := r.fetchHTTPRoute(ctx, rlp)
+	if err != nil {
+		// The object should also exist
+		return err
+	}
 
-		if err := updateEnvoyFilter(ctx, wasmEnvoyFilter, rlp, gwKey, gwLabels); err != nil {
-			logger.Error(err, "failed to create/update EnvoyFilter containing wasm filters")
-			return err
-		}
+	// Reconcile the back reference:
+	httpRouteAnnotations := httpRoute.GetAnnotations()
+	if httpRouteAnnotations == nil {
+		httpRouteAnnotations = map[string]string{}
+	}
 
-		if err := r.addParentRefEntry(ctx, wasmEnvoyFilter, owner); err != nil {
-			logger.Error(err, "failed to add ownerRLP entry to the ratelimit filters patch")
+	rlpKey := client.ObjectKeyFromObject(rlp)
+	val, ok := httpRouteAnnotations[common.RateLimitPolicyBackRefAnnotation]
+	if ok {
+		if val != rlpKey.String() {
```

Comment:
Not an issue but a good point to refer in docs that we don't support multiple RLP to target single route at the moment.